docker_images/diffusers/app/pipelines/text_to_image.py

```diff
@@ -70,29 +60,27 @@ def __init__(self, model_id: str):
 
         if model_type == "LoraModel":
             model_to_load = model_data.cardData["base_model"]
+            self.model_id = model_to_load
             if not model_to_load:
                 raise ValueError(
                     "No `base_model` found. Please include a `base_model` on your README.md tags"
                 )
-            if model_to_load == "stabilityai/stable-diffusion-xl-base-1.0":
-                vae = AutoencoderKL.from_pretrained(
-                    "madebyollin/sdxl-vae-fp16-fix",
-                    torch_dtype=torch.float16,  # load fp16 fix VAE
-                )
-                kwargs["vae"] = vae
-                kwargs["variant"] = "fp16"
 
-            self.ldm = DiffusionPipeline.from_pretrained(
-                model_to_load, use_auth_token=use_auth_token, **kwargs
-            )
-            weight_name = file_to_load if not is_diffusers_lora else None
+            weight_name = self._get_lora_weight_name(model_data)
+            self._load_sd_with_sdxl_fix(model_to_load, **kwargs)
             self.ldm.load_lora_weights(
-                model_id, weight_name=weight_name, use_auth_token=use_auth_token
+                model_id, weight_name=weight_name, use_auth_token=self.use_auth_token
             )
+            self.current_lora_adapter = model_id
         else:
-            self.ldm = AutoPipelineForText2Image.from_pretrained(
-                model_id, use_auth_token=use_auth_token, **kwargs
-            )
+            # TODO/TBD: is there any specific reason not to always apply it ?
```

Comment:
If anyone can anwser this :) (maybe @apolinario )

In the previous version, the sdxl vae fix was only applied in the lora case, is there any specific reason not to apply the fix on the base model as well ? (even when not using lora)

This would avoid the additional SDXL_FP16_FIX and making a difference between a stabilityai/stable-diffusion-xl-base-1.0 deployment meant to be used with lora adapters and the same model meant to be used as a standalone one

Reply:
We can apply it in all cases, not only the LoRA cases

Reply:
(As long as the base model is SDX. For SDv1-5 or 2-1, this wouldn't work)

Reply:
done